lighthouse-core/gather/gatherers/fonts.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+const fontFaceDescriptors = [
+  'display',
+  'family',
+  'featureSettings',
+  'stretch',
+  'style',
+  'unicodeRange',
+  'variant',
+  'weight',
+];
+
+/* eslint-env browser*/
+
+/**
+ * Collect applied webfont data from `document.fonts`
+ * @return {{}}
+ */
+function getAllLoadedFonts() {
+  const getFont = fontFace => {
+    const fontRule = {};
+    fontFaceDescriptors.forEach(descriptor => {
+      fontRule[descriptor] = fontFace[descriptor];
+    });
+
+    return fontRule;
+  };
+
+  return document.fonts.ready.then(() => {
+    return Array.from(document.fonts).filter(fontFace => fontFace.status === 'loaded')
+      .map(getFont);
+  });
+}
+
+/**
+ * Collect authored webfont data from the `CSSFontFaceRule`s present in document.styleSheets
+ * @return {{}}
+ */
+function getFontFaceFromStylesheets() {
```

Comment:
let's throw the `/* istanbul ignore next */` on these like paul just did in #4396